nixos/modules/programs/shadow.nix

```diff
@@ -1,129 +1,210 @@
+          GROUP=100
+          HOME=/home
+          SHELL=${utils.toShellPath config.users.defaultUserShell}
+        '';
       };
 
-    security.pam.services =
-      { chsh = { rootOK = true; };
-        chfn = { rootOK = true; };
-        su = { rootOK = true; forwardXAuth = true; logFailures = true; };
-        passwd = {};
-        # Note: useradd, groupadd etc. aren't setuid root, so it
-        # doesn't really matter what the PAM config says as long as it
-        # lets root in.
-        useradd = { rootOK = true; };
-        usermod = { rootOK = true; };
-        userdel = { rootOK = true; };
-        groupadd = { rootOK = true; };
-        groupmod = { rootOK = true; };
-        groupmems = { rootOK = true; };
-        groupdel = { rootOK = true; };
-        login = { startSession = true; allowNullPassword = true; showMotd = true; updateWtmp = true; };
-        chpasswd = { rootOK = true; };
+    security.pam.services = {
+      chsh = { rootOK = true; };
+      chfn = { rootOK = true; };
+      su = {
+        rootOK = true;
+        forwardXAuth = true;
+        logFailures = true;
       };
-
-    security.wrappers = {
-      su        = mkSetuidRoot "${pkgs.shadow.su}/bin/su";
-      sg        = mkSetuidRoot "${pkgs.shadow.out}/bin/sg";
-      newgrp    = mkSetuidRoot "${pkgs.shadow.out}/bin/newgrp";
-      newuidmap = mkSetuidRoot "${pkgs.shadow.out}/bin/newuidmap";
-      newgidmap = mkSetuidRoot "${pkgs.shadow.out}/bin/newgidmap";
-    } // lib.optionalAttrs config.users.mutableUsers {
-      chsh   = mkSetuidRoot "${pkgs.shadow.out}/bin/chsh";
-      passwd = mkSetuidRoot "${pkgs.shadow.out}/bin/passwd";
+      passwd = { };
+      # Note: useradd, groupadd etc. aren't setuid root, so it
+      # doesn't really matter what the PAM config says as long as it
+      # lets root in.
+      useradd = { rootOK = true; };
+      usermod = { rootOK = true; };
+      userdel = { rootOK = true; };
+      groupadd = { rootOK = true; };
+      groupmod = { rootOK = true; };
+      groupmems = { rootOK = true; };
+      groupdel = { rootOK = true; };
+      login = {
+        startSession = true;
+        allowNullPassword = true;
+        showMotd = true;
+        updateWtmp = true;
+      };
+      chpasswd = { rootOK = true; };
     };
+
+    security.wrappers =
+      let
+        mkSetuidRoot = source: {
+          setuid = true;
+          owner = "root";
+          group = "root";
+          inherit source;
+        };
+      in
+      {
+        su = mkSetuidRoot "${pkgs.shadow.su}/bin/su";
+        sg = mkSetuidRoot "${pkgs.shadow.out}/bin/sg";
+        newgrp = mkSetuidRoot "${pkgs.shadow.out}/bin/newgrp";
+        newuidmap = mkSetuidRoot "${pkgs.shadow.out}/bin/newuidmap";
+        newgidmap = mkSetuidRoot "${pkgs.shadow.out}/bin/newgidmap";
+      } // optionalAttrs config.users.mutableUsers {
+        chsh = mkSetuidRoot "${pkgs.shadow.out}/bin/chsh";
+        passwd = mkSetuidRoot "${pkgs.shadow.out}/bin/passwd";
```

Comment:
Should we add a package option?

Reply:
Could you elaborate more?

Reply:
see https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/programs/nexttrace.nix#L11